libs/simdvec/src/test/java/org/elasticsearch/simdvec/ESVectorUtilTests.java

```diff
@@ -51,7 +61,7 @@ public void testIpFloatByte() {
         for (int i = 0; i < q.length; i++) {
             expected += q[i] * d[i];
         }
-        assertEquals(expected, ESVectorUtil.ipFloatByte(q, d), 1e-6);
+        assertEquals(expected, impl.applyAsDouble(q, d), 1e-2);
```

Comment:
its interesting that the error is so much higher. I wouldn't expect it to be so much worse like this.

Reply:
Yes, that surprised me too. I saw the same in the lucene vector operations PR. It's most likely due to differences in float addition ordering - but the delta is higher than I would like. Is it worth further investigation, or is this acceptable?

Reply:
My concern @thecoop is that we have a fencepost error where we accidentally drop dimensions (e.g. skip the tail). I am thinking that a 1e-2 might actually allow this type of fencepost (e.g. forgetting the last float that is actually tiny).

Reply:
I've added a change to scale the delta according to the vector size - the largest differences are associated with large vector sizes (as you might expect), this should use a tight bound for small vectors, and relaxing it for large, so we can check we're not missing any dims with small vectors